streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3302,6 +3304,91 @@ object ZStream extends ZStreamPlatformSpecificConstructors with Serializable {
   def fromJavaIteratorManaged[R, E, A](iterator: ZManaged[R, E, ju.Iterator[A]]): ZStream[R, E, A] =
     managed(iterator).flatMap(StreamEffect.fromJavaIterator)
 
+  /**
+   * Creates a managed (impure) Iterator from a Stream
+   */
+  def toIterator[R, E, A](
+    stream: ZStream[R, E, A],
+    afterLastPull: URIO[R, Any] = ZIO.unit
```

Comment:
What if we get rid of the concept of `afterLastPull` here? We are already creating this in the context of a `ZManaged`, so once we exit the scope of the `ZManaged` the stream's finalizers will automatically be executed. If the caller wants to take some further action when the stream is empty that can do it at that point.

Reply:
~We can, but! The initial use case need it to be able to unsafely turn a  `ZManaged[_,_,Iterator[A]]` into a `ZIO[_,_,Iterator[A]]`.~

I just found another way to do it! Let's remove it.